tests/run-tests.py

```diff
@@ -999,9 +999,21 @@ def run_one_test(test_file):
             with open(filename_mupy, "wb") as f:
                 f.write(output_mupy)
             failed_tests.append((test_name, test_file))
+            if uses_unittest:
+                rm_f(filename_expected)  # in case left over from previous failed run
```

Comment:
It seems that `uses_unittest` is True is equivalent to `output_expected` is None.  So might be slightly better logic to put this up above as:
```py
if output_expected is not None:
    with open(filename_expected, "wb") as f:
        f.write(output_expected)
else:
    rm_f(filename_expected)
```

Reply:
Ah yes, that's much cleaner. Thanks!